tests/test_views_permissions.py

```diff
@@ -50,16 +53,51 @@ def setUp(self):
     def test_permissions_list_entries_for_everyone(self):
         resp = self.app.get('/permissions', headers=self.everyone_headers)
         buckets = resp.json['data']
-        self.assertEqual(len(buckets), 5)
+        self.assertEqual(len(buckets), 6)
+
+        uris = [bucket['uri'] for bucket in buckets]
+        self.assertEqual(sorted(uris), [
+            '/buckets',
+            '/buckets/beers',
+            '/buckets/beers/collections/barley',
+            '/buckets/beers/collections/barley/records/{}'.format(RECORD_ID),
+            '/buckets/beers/groups/amateurs',
+            '/buckets/beers/groups/barley',
+        ])
 
     def test_bucket_create_permission_exists(self):
         resp = self.app.get('/permissions', headers=self.everyone_headers)
         buckets = resp.json['data']
-        toplevel_bucket = buckets[4]
-        self.assertEqual(toplevel_bucket, {
+        toplevel_bucket = [b for b in buckets if b['uri'] == '/buckets']
+        self.assertEqual(toplevel_bucket, [{
             'permissions': ['bucket:create'],
             'resource_name': 'bucket',
-            'uri': '/buckets'})
+            'uri': '/buckets'}])
+
+    def test_permissions_can_be_paginated(self):
+        """Verify that pagination doesn't squash same IDs"""
+        # kinto_http.js uses this sort of request when listing permissions
+        real_permissions = self.app.get('/permissions', headers=self.everyone_headers).json['data']
+
+        def sort_by_uri(l):
+            return sorted(l, key=lambda r: r['uri'])
+
+        for order in ['id', '-id', 'uri', '-uri']:
```

Comment:
I think it is better to use `pytest` parametrize function for this (better failure messages etc)

Reply:
This is a `unittest` flavored test, which doesn't have any concept of "parametrization". I can convert it to a `pytest` style test, but it seems a bit aggressive of a change. Alternately I can use metaprogramming to generate all the tests, but it seems a bit ugly. In the meantime I used `subTest`, which is a little bit like `parametrize` in that it annotates the test failure with the values that were being tested. What do you think?

Reply:
You're right, leave it like this...